cms/categories/views.py

```diff
@@ -1,3 +1,38 @@
 from django.shortcuts import render
+from django.views.generic import ListView, DetailView
 
-# Create your views here.
+from cms.categories.models import Category, CategoryPage
+
+
+class CategoryListView(ListView):
+    model = Category
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["self"] = {"title": "All categories"}
+        return context
+
+
+class CategoryDetailView(DetailView):
+    model = Category
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        pages = CategoryPage.objects.live()
+        category = context["object"]
+        pages = pages.filter(categorypage_category_relationship__category=category)
+        ordered_pages = pages.order_by("-last_published_at")
+        page_data = [
```

Comment:
I'm happy with this duplication. If we find ourselves doing this again, we might want to consider refactoring.
(comment only)